### PR TITLE
Expose configuration and user on SyncSession.

### LINKED ISSUE #431: SyncSession - Utility methods
Expose the various utility methods we also have in Realm Java:

```java
interface SyncSession {
  val configuration: SyncConfiguration
  val user: User
  val serverUrl: String
}
}
```

### PR DESCRIPTION
**Pull request by elle-j**
## Description

Closes #431.

Makes the `configuration` and `user` available on `SyncSession`.

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)

### CODE REVIEW

File: packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt [hunk 1]
```diff
@@ -73,6 +74,21 @@ public interface SyncSession {
...
+
+    /**
+     * The [User] defined by the [SyncConfiguration] to authenticate the session on Atlas App Services.
+     */
+    public val user: User
+
+    /**
+     * The URL describing the remote Realm that this session connects to and synchronizes changes with.
+     */
+    public val serverUrl: String
```
Comment (nirinchev) [inline packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:90]: Not sure we should be exposing this - for pretty much all users, this will be `realm.mongodb.com` and the internal users that absolutely need to look it up can go through the user.

Comment (cmelchior) [reply packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:90]: Good point. This is probably a leftover from the pre-v10 API's. But you are correct, there is no reason we need to expose this in Kotlin. I just forgot to think more closely about it when creating this issue.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * Added support for reverse relationships through the `linkingObjects` delegate. See the function documentation for more details. (Issue [#1021](https://github.com/realm/realm-kotlin/pull/1021))
 * Added support for `BsonObjectId` and its typealias `org.mongodb.kbson.ObjectId` as a replacement for `ObjectId`. `io.realm.kotlin.types.ObjectId` is still functional but has been marked as deprecated.
 * [Sync] Added support for `BsonObjectId` as partition value.
+* [Sync] Exposed `configuration`, `user`, and `serverUrl` on `SyncSession`. (Issue [#431](https://github.com/realm/realm-kotlin/issues/431))
```
Comment (cmelchior) [inline CHANGELOG.md:12]: 🥇

File: packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt [hunk 3]
```diff
@@ -496,6 +496,30 @@ class SyncSessionTests {
         }
     }
 
+    @Test
+    fun session_getConfiguration() {
```
Comment (cmelchior) [inline packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt:500]: We should also test this inside the error handler so we capture the exception

File: packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt [hunk 4]
```diff
@@ -73,6 +74,21 @@ public interface SyncSession {
      */
     public val state: State
 
+    /**
+     * The [SyncConfiguration] responsible for controlling the session.
+     */
+    public val configuration: SyncConfiguration
+
+    /**
+     * The [User] defined by the [SyncConfiguration] to authenticate the session on Atlas App Services.
```
Comment (cmelchior) [inline packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:83]:
```suggestion
     * The [User] used to authenticate the session on Atlas App Services.
```

I would reword this slightly to just include the most important bits.

**Review by cmelchior — CHANGES_REQUESTED** [review]
Great start 💯... There is a few minor comments + we should also find a way to test this inside the SyncConfiguration error handler.

File: packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt (hunk 1 above)
Comment (elle-j) [reply packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:90]: Ah okay! I'll revert that one 👍

Comment (nhachicha) [reply packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:90]: The only use case I can think of is testing, If we spin up a local BAAS?

File: packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt [hunk 5]
```diff
@@ -73,6 +74,16 @@ public interface SyncSession {
      */
     public val state: State
 
+    /**
+     * The [SyncConfiguration] responsible for controlling the session.
```
Comment (clementetb) [inline packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:78]: We could use a comment explaining in what conditions will it throw.

**Review by clementetb — APPROVED** [review]
Looking good to me, added a minor comment 🚀.

File: packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt [hunk 6]
```diff
@@ -496,6 +499,69 @@ class SyncSessionTests {
...
+            schema = setOf(ChildPk::class),
+            user = user,
+            partitionValue = partitionValue
+        )
+            .name("test1.realm")
+            .build()
+        val realm1 = Realm.open(config1)
+        assertNotNull(realm1)
+
+        // Open another realm with the same entity but change the type of a field
```
Comment (cmelchior) [inline packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt:528]: Good way of testing this 👍

**Review by cmelchior — APPROVED** [review]
💯 👍

File: packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt (hunk 5 above)
Comment (elle-j) [reply packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt:78]: Suggestion:
```kt
/**
 * The [SyncConfiguration] responsible for controlling the session.
 * 
 * @throws IllegalStateException if accessed from inside a [SyncSession.ErrorHandler] due to session errors.
 */
```
